### PR TITLE
Issue fix #42 & #43

### LINKED ISSUE #42: Handle special characters in password
``` bash
» afpv2 show account-name
Password for mriehl: 
Traceback (most recent call last):
  File "/usr/local/bin/afpv2", line 9, in <module>
    load_entry_point('afp-cli==1.3.1.post166', 'console_scripts', 'afpv2')()
  File "/usr/local/lib/python2.7/dist-packages/afp_cli/cliv2.py", line 59, in main
    unprotected_main()
  File "/usr/local/lib/python2.7/dist-packages/afp_cli/cliv2.py", line 106, in unprotected_main
    role = arguments['<rolename>'] or get_first_role(federation_client, account)
  File "/usr/local/lib/python2.7/dist-packages/afp_cli/cli_functions.py", line 48, in get_first_role
    accounts_and_roles = federation_client.get_account_and_role_list()
  File "/usr/local/lib/python2.7/dist-packages/afp_cli/client.py", line 43, in get_account_and_role_list
    accounts_and_roles = self.call_api("/account")
  File "/usr/local/lib/python2.7/dist-packages/afp_cli/client.py", line 30, in call_api
    self._password))
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 67, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 53, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 454, in request
    prep = self.prepare_request(req)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 388, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 297, in prepare
    self.prepare_auth(auth, url)
  File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 490, in prepare_auth
    r = auth(self)
  File "/usr/local/lib/python2.7/dist-packages/requests/auth.py", line 51, in __call__
    r.headers['Authorization'] = _basic_auth_str(self.username, self.password)
  File "/usr/local/lib/python2.7/dist-packages/requests/auth.py", line 31, in _basic_auth_str
    b64encode(('%s:%s' % (username, password)).encode('latin1')).strip()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 8: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by karolyi**
I am the wonderfully reorganized stack of commits from the chaos that emerged with a bad constellation under travis and cram:

https://github.com/brodie/cram/issues/7

Please merge me, so the world can be a better place.

In other news, I happen to fix 2 bugs at the same time: #42 and #43.

### CODE REVIEW

**Comment by karolyi** [comment]
@esc ?

**Comment by esc** [comment]
Was in a meeting, sry.

File: .travis.yml [hunk 1]
```diff
@@ -6,13 +6,15 @@ python:
...
   - "pypy"
-  - "pypy3"
 
 install:
-  - pip install pybuilder
-  - pip install coveralls
-  - pyb install_dependencies
-  
+  - travis_retry pip install -U pip
+  - travis_retry pip install git+https://github.com/karolyi/pybuilder.git cram coveralls
```
Comment (esc) [inline .travis.yml]: Don't think we need to install cram here.

Comment (esc) [inline .travis.yml]: Also, can you link me the issue in pybuilder that fixes the cram plugin?

File: .travis.yml [hunk 2]
```diff
@@ -6,13 +6,15 @@ python:
...
+  - travis_retry pyb install_dependencies
+
+env:
+  PYTHONIOENCODING=UTF-8
```
Comment (esc) [inline .travis.yml:15]: Is this really required, or just for good measure?

**Comment by esc** [comment]
I rebased on current master.

**Comment by karolyi** [comment]
https://github.com/pybuilder/pybuilder/pull/316

**Comment by esc** [comment]
Sorry, ran into password reset troubles, will continue tomorrow.

**Comment by esc** [comment]
I set myself a new password with an Umlaut and afp-cli was fine, but the `afp-core` on the other side didn't like it, I think. I am not sure though, because then I couldn't log into my local Linux machine anymore. Although that may have another reason.

I'd like to hold off merging this until it is confirmed to work, of which I am not yet convinced.

**Comment by karolyi** [comment]
rewritten, again.

only role name parameter can be passed with UTF-8 characters now, which is part of the url, so it gets encoded.

**Comment by karolyi** [comment]
fixes issues #42  and #43

File: src/main/python/afp_cli/client.py [hunk 3]
```diff
@@ -1,41 +1,50 @@
...
+        self.password = kwargs.get('password', '')
+
         self.api_url = kwargs.get('api_url', None)
         self.ssl_verify = kwargs.get('ssl_verify', True)
 
     def call_api(self, url_suffix):
         """Send a request to the aws federation proxy"""
+        url_orig = '{0}{1}'.format(self.api_url, url_suffix)
+        # Workaround: request seems to mystically fail with utf8 stuff
+        url = requests.utils.requote_uri(url_orig)
```
Comment (esc) [inline src/main/python/afp_cli/client.py:32]: Can we remove this line now, or should we leave it in for good measure? If so, perhaps adapt the comment.

File: src/main/python/afp_cli/compat.py [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
+
 try:
-    from ordereddict import OrderedDict  # NOQA
+    from ordereddict import OrderedDict
 except ImportError:  # pragma: no cover
-    from collections import OrderedDict  # NOQA
+    from collections import OrderedDict
+
+# Linting purposes, no other use
+OrderedDict
```
Comment (esc) [inline src/main/python/afp_cli/compat.py]: Let's do a `# flake8: noqa` and `# pragma: no cover` for the whole file? Then we can remove this.

File: src/main/python/afp_cli/client.py (hunk 3 above)
Comment (karolyi) [inline src/main/python/afp_cli/client.py:32]: yeah, maybe remove the comment but I'd definitely keep the variable, so it won't have to be compiled at the parameter handover time.

actually there was more that was removed, I just run through the whole source code and removed the non relevant parts, so this is what remained.

**Comment by esc** [comment]
It looks good, only a few minor nitpicks.

Also, we have had some problems with the `unicode_literals` in the past in the afp project, so we should be vigilant for errors that may be caused by that future import.

File: src/main/python/afp_cli/client.py (hunk 3 above)
Comment (esc) [inline src/main/python/afp_cli/client.py:32]: Yeah, we should definately keep `url_orig`.

File: src/main/python/afp_cli/compat.py (hunk 4 above)
Comment (karolyi) [inline src/main/python/afp_cli/compat.py]: sure thing, it's my editor again that won't notice the #NOQA statements

**Pull request merged**